consensus/src/simplex/actors/voter/actor.rs

```diff
@@ -319,12 +299,21 @@ impl<
     }
 
     /// Tracks a verified nullification certificate if it is new.
-    async fn handle_nullification(&mut self, nullification: Nullification<S>) {
+    async fn handle_nullification(
+        &mut self,
+        nullification: Nullification<S>,
+    ) -> Option<Voter<S, D>> {
         let view = nullification.view();
         let msg = Voter::Nullification(nullification.clone());
-        if self.state.add_verified_nullification(nullification) {
-            self.append_journal(view, msg).await;
+
+        // Add verified nullification to journal
+        if !self.state.add_verified_nullification(nullification) {
+            return None;
         }
+        self.append_journal(view, msg).await;
+
+        // If we were the proposer, we should emit the notarization that we built our proposal on
+        self.state.emit(view)
```

Comment:
nit: consider doing `if a { append; emit; }` instead of `if !a { return }; append; emit;`

Reply:
I preferred the shallow if here.